src/main/java/com/rocs/infirmary/application/service/student/profile/impl/StudentHealthProfileServiceImpl.java

```
@@ -0,0 +1,98 @@
+package com.rocs.infirmary.application.service.student.profile.impl;
+
+import com.rocs.infirmary.application.domain.medical.history.MedicalHistory;
+import com.rocs.infirmary.application.domain.person.Person;
+import com.rocs.infirmary.application.domain.section.Section;
+import com.rocs.infirmary.application.domain.student.Student;
+import com.rocs.infirmary.application.domain.student.dto.student.heath.profile.CreateStudentHealthProfileRequestDTO;
+import com.rocs.infirmary.application.repository.medical.history.MedicalHistoryRepository;
+import com.rocs.infirmary.application.repository.student.StudentRepository;
+import com.rocs.infirmary.application.service.student.profile.StudentHealthProfileService;
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class StudentHealthProfileServiceImpl implements StudentHealthProfileService {
+
+   private StudentRepository studentRepository;
+   private MedicalHistoryRepository medicalHistoryRepository;
+    @Autowired
+    public StudentHealthProfileServiceImpl(StudentRepository studentRepository, MedicalHistoryRepository medicalHistoryRepository) {
+        this.studentRepository = studentRepository;
+        this.medicalHistoryRepository = medicalHistoryRepository;
+    }
+    @Override
+    @Transactional
+    public Student createStudentHealthProfile(CreateStudentHealthProfileRequestDTO dto) {
+
+            validateRequiredFields(dto);
+
+            if(studentRepository.existsByLrn(dto.getLrn())){
+                throw new IllegalArgumentException("Student with LRN " + dto.getLrn() + " already exists");
```

use LOGGER.error and if there is a need to throw an exception, handle it using exception handler

refer to sir Donnie's recording